openmc/bounding_box.py

```diff
@@ -35,17 +35,24 @@ class BoundingBox(tuple):
         The width of the x, y and z axis in [cm]
     """
 
-    def __new__(cls, lower_left: Iterable[float], upper_right: Iterable[float]):
+    def __init__(self, lower_left: Iterable[float], upper_right: Iterable[float]):
         check_length("lower_left", lower_left, 3, 3)
         check_length("upper_right", upper_right, 3, 3)
-        lower_left = np.array(lower_left, dtype=float)
-        upper_right = np.array(upper_right, dtype=float)
-        return tuple.__new__(cls, (lower_left, upper_right))
+        self._bounds = np.asarray([lower_left, upper_right], dtype=float)
 
     def __repr__(self) -> str:
         return "BoundingBox(lower_left={}, upper_right={})".format(
             tuple(self.lower_left), tuple(self.upper_right))
 
+    def __getitem__(self, key) -> np.ndarray:
+        return self._bounds[key]
+
+    def __setitem__(self, key, val):
+        self._bounds[key]
+
+    def __hash__(self) -> int:
+        return hash(self._bounds.data)
```

Comment:
If you are making this object mutable, it probably doesn't make sense for it to be hashable too. Unless we really expect bounding boxes to be used in sets or as dictionary keys, I would remove this.